File: backend-java/comment-service/src/main/java/be/pxl/services/services/CommentService.java
package be.pxl.services.services;


import be.pxl.services.domain.Comment;
import be.pxl.services.domain.Post;
import be.pxl.services.domain.User;
import be.pxl.services.exception.CommentNotFoundException;
import be.pxl.services.repository.CommentRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class CommentService {
    private final CommentRepository commentRepository;

    public Comment addComment(Long postId, Long userId, String content) {
        Comment comment = new Comment();
        comment.setContent(content);
        comment.setPost(new Post(postId));  // Aangenomen dat je een Post-object hebt
        comment.setUser(new User(userId));  // Aangenomen dat je een User-object hebt
        return commentRepository.save(comment);
    }

    public List<Comment> getCommentsByPost(Long postId) {
        return commentRepository.findByPostId(postId);
    }

    public Comment updateComment(Long commentId, Long userId, String newContent) throws CommentNotFoundException {
        Optional<Comment> optionalComment = commentRepository.findByIdAndUserId(commentId, userId);
        if (optionalComment.isPresent()) {
            Comment comment = optionalComment.get();
            comment.setContent(newContent);
            return commentRepository.save(comment);
        } else {
            throw new CommentNotFoundException("Comment not found or not owned by user");
        }
    }

    public void deleteComment(Long commentId, Long userId) throws CommentNotFoundException {
        Optional<Comment> optionalComment = commentRepository.findByIdAndUserId(commentId, userId);
        if (optionalComment.isPresent()) {
            commentRepository.delete(optionalComment.get());
        } else {
            throw new CommentNotFoundException("Comment not found or not owned by user");
        }
    }
}

File: frontend-web/frontend-web/src/app/shared/services/comment.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../../environments/environment';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CommentService {

  private apiUrl = environment.apiUrl + 'comment/api/comment';

  constructor(private http: HttpClient) { }

    // Comment toevoegen
    addComment(postId: number, userId: number, content: string): Observable<Comment> {
      return this.http.post<Comment>(`${this.apiUrl}/post/${postId}/user/${userId}`, content);
    }
  
    // Reacties van een post ophalen
    getCommentsByPost(postId: number): Observable<Comment[]> {
      return this.http.get<Comment[]>(`${this.apiUrl}/post/${postId}`);
    }

    // Reactie bewerken
  updateComment(commentId: number, userId: number, newContent: string): Observable<Comment> {
    return this.http.put<Comment>(`${this.apiUrl}/${commentId}/user/${userId}`, newContent);
  }

  // Reactie verwijderen
  deleteComment(commentId: number, userId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${commentId}/user/${userId}`);
  }
}
